### PR TITLE
docs(customizing): add guide for overriding styles

### PR DESCRIPTION
**Pull request by willshowell**
Fixes https://github.com/angular/material2/issues/6087

### CODE REVIEW

File: guides/customizing-component-styles.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+2. Are your styles more specific than the defaults?
+3. Is the component a child of your component, or does it exist elsewhere in the DOM?
+
+##### View encapsulation
+
+By default, Angular component styles are scoped to affect the component's view. This means that
+the styles you write will affect all the elements in your component template. They will *not*
+affect elements that are children of other components within your template. You can read more
+about view encapsulation in the
+[Angular documentation](https://angular.io/guide/component-styles#view-encapsulation).
```
Comment (jelbourn) [inline guides/customizing-component-styles.md]: Would be good to include a link to the recent blog post on this subject as well: https://blog.angular.io/the-state-of-css-in-angular-4a52d4bd2700
(which was partially ghostwritten by me)

**Comment by willshowell** [comment]
Done! Thank you and Stephen for putting that post together

File: guides/customizing-component-styles.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Each CSS declaration has a level of *specificity* based on the type and number of selectors used.
+More specific styles will take precedence over less specific styles. Angular Material uses the
+least specific selectors possible for its components in order to make it easy to override them.
+You can read more about specificity and how it is calculated on the
+[MDN web docs](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity).
+
+##### Component location
+
+Some Angular Material components, specifically overlay-based ones like MdDialog, MdSnakcbar, etc.,
```
Comment (kara) [inline guides/customizing-component-styles.md]: Typo: `MdSnakvbar` -> `MdSnackbar`

**Comment by kara** [comment]
@willshowell Thanks for the quick fix!

**Pull request merged**